autogen.sh

```diff
@@ -2,6 +2,8 @@
 
 test -d ./config || mkdir ./config
 
+test -f modules/test-sos/configure.ac || { echo -n "WARNING: The test directory is empty. Please run the following command to install the SOS tests:\ngit submodule update --init"; }
```

Comment:
The warning output doesn't look right to me on my system.  It's ok to have a few instances of `echo` to force the newlines.

Also, let's just `exit 1` if the submodule is not there because the whole build will fail in that case as things are now.

Reply:
If we're going to be exiting should we also upgrade the "Warning" to an "Error"?

Reply:
Yes please, thanks.